### ISSUE TITLE
rhub_check() fails with vignette

### ISSUE THREAD

**Issue by MatthieuStigler** [r-hub/rhub#643]
This is very similar to #629  (but opening a new issue since 629 was closed). 

Running `rhub_check()` will systematically fail on various platforms (tested so far: "linux", "windows", "macos", "atlas"). 

Typical message will be ([link](https://github.com/MatthieuStigler/tsDyn/actions/runs/11530270761/job/32099644082)):

> Running 'texi2dvi' on 'ThCointOverview.tex' failed.
Messages:
You don't have a working TeX binary (tex) installed anywhere in
your PATH, and texi2dvi cannot proceed without one.  If you want to use
this script, you'll need to install TeX

or ([link](https://github.com/MatthieuStigler/tsDyn/actions/runs/11524976526/job/32086285449)):

> texi2dvi script/program not available, using emulation
Error: processing vignette 'tsDyn.Stex' failed with diagnostics:
pdflatex is not available
--- failed re-building ‘tsDyn.Stex’

is it possible to either add an option to skip vignettes (`--no-build-vignettes`? `--no-vignettes`?) or add some pdflatex binaries on the platforms? or am I doing something wrong here?

Thanks!

**Comment by sashahafner** [comment]
Have you found a solution?

**Comment by MatthieuStigler** [comment]
so far no :-(

**Comment by sashahafner** [comment]
I just now have something that works.
See the addition to .github/workflows/rhub.yaml here and below (L64+, L100+):
https://github.com/sashahafner/biogas/blob/f6dabc6a51db5e7ec9f48290ebdc7c0c0dc9feef/.github/workflows/rhub.yaml#L64

```
  linux-containers:
...
    steps:
...
      - name: Setup TeX Live
        uses: teatimeguest/setup-texlive-action@v3
        with:
          packages:  |
            scheme-medium
            xcolor
            framed       
```

Based on this solution:
https://github.com/r-hub/rhub/issues/583#issuecomment-2030225442

**Comment by jdblischak** [comment]
> is it possible to either add an option to skip vignettes (`--no-build-vignettes`? `--no-vignettes`?)

Yes. I found a way. There are no `inputs` to the action to do this (the args you pass with `with:`). However, looking at the source code of [`r-hub/actions/check`](https://github.com/r-hub/actions/blob/main/check/action.yaml), I noticed there is a env var [`CHECK_ARGS`](https://github.com/r-hub/actions/blob/d0897441695fddb724d427f97cb904529e08e774/check/action.yaml#L123).

I was able to completely ignore the vignettes by defining the following env vars:

```yaml
      - uses: r-hub/actions/run-check@v1
        env:
          BUILD_ARGS: --no-build-vignettes
          CHECK_ARGS: --ignore-vignettes
```

Note that you need to define this twice in `rhub.yaml` if you want to test in Linux and non-Linux environments.

**Comment by jdblischak** [comment]
> I just now have something that works.

This solution didn't work for me. That step successfully installed the missing LaTeX packages, but the vignette builds still failed with missing packages errors. Below is what I had:

```yaml
      - name: Setup TeX Live
        uses: teatimeguest/setup-texlive-action@v3
        with:
          packages: |
            amsmath
            color
            fancyvrb
            float
            framed
            geometry
            grfext
            hyperref
            inputenc
            parskip
```

But I still got the errors:

```
! LaTeX Error: File `grfext.sty' not found.

! LaTeX Error: File `parskip.sty' not found.
```
